RegexExceptionHandling/RegexExceptionHandling/PhoneValidator.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Text.RegularExpressions;
+namespace RegexExceptionHandling
+{
+    /// <summary>
+    /// This class containes all methods used to validate phone numbers via regular expressions.
+    /// </summary>
+    public class PhoneValidator
+    {
+        /// <summary>
+        /// This method validates a single phone number using regular expressions.
+        /// </summary>
+        /// <param name="numberToValidate"></param>
+        /// <returns></returns>
+        public bool validatePhoneNumber(string numberToValidate)
+        {
+            // This regular expression validates 10 & 11 digit US telephone numbers.
+            Regex phoneNumberExpression7 =
+                new Regex(@"^(?:\+?1[-/. ]?)?\(?([0-9]{3})\)?[-/. ]?([0-9]{3})[-/. ]?([0-9]{4})$");
```

Comment:
Good job with this pattern. The only issue my test set found was that area codes with only a leading or training parenthesis would still be considered valid:

(444 333-2222
444) 333-2222

(-1.0)